### PR TITLE
Make use of promised dependencies

### PR DESCRIPTION
**Pull request by jviotti**

### CODE REVIEW

File: lib/models/application.coffee [hunk 1]
```diff
@@ -274,10 +248,11 @@ exports.remove = (name, callback) ->
 ###
 exports.restart = (name, callback) ->
 	exports.get(name).then (application) ->
-		return request.requestAsync
+		return request.send
 			method: 'POST'
 			url: "/application/#{application.id}/restart"
-	.nodeify(_.unary(callback))
+	.return(undefined)
```
Comment (abresas) [inline lib/models/application.coffee]: why undefined?

File: tests/models/device.spec.coffee [hunk 2]
```diff
@@ -14,276 +13,180 @@ describe 'Device Model:', ->
...
 
-				it 'should return false', (done) ->
-					device.has 'MyDevice', (error, hasDevice) ->
-						expect(error).to.not.exist
-						expect(hasDevice).to.be.false
-						done()
+			it 'should add application_name', (done) ->
+				device.get 'MyDevice', (error, device) =>
+					expect(device.application_name).to.equal(@device.application[0].app_name)
+					done()
```
Comment (abresas) [inline tests/models/device.spec.coffee]: I usually do .nodeify(done) on tests

**Comment by abresas** [comment]
lgtm, left some notes

File: lib/models/application.coffee (hunk 1 above)
Comment (jviotti) [inline lib/models/application.coffee]: To prevent resolving the promise with the request response, just as a way to avoid polluting the API with unnecessary information.

File: tests/models/device.spec.coffee (hunk 2 above)
Comment (jviotti) [inline tests/models/device.spec.coffee]: I plan to move the SDK tests to use `chai-as-promised` as the other modules, but plan to do it in another PR.

File: lib/auth.coffee [hunk 3]
```diff
@@ -173,8 +170,7 @@ exports.isLoggedIn = (callback) ->
 #		console.log(token)
 ###
 exports.getToken = (callback) ->
-	Promise.try ->
-		savedToken = token.get()
+	token.get().then (savedToken) ->
```
Comment (Page-) [inline lib/auth.coffee]: This could be a `.tap`

File: lib/auth.coffee [hunk 4]
```diff
@@ -201,8 +197,7 @@ exports.getToken = (callback) ->
 #		console.log(id)
 ###
 exports.getUserId = (callback) ->
-	Promise.try ->
-		id = token.getUserId()
+	token.getUserId().then (id) ->
```
Comment (Page-) [inline lib/auth.coffee]: This could be a `.tap`
